### PR TITLE
Preview file changes during Merge/Publish for dev deployments

### PR DESCRIPTION
**Pull request by k-anshul**
https://linear.app/rilldata/issue/ENG-1120/option-to-view-file-changes-before-committing-in-cloud-editing

**Note**: Option to view full file diff will be raised in a follow up 

**Checklist:**
- [ ] Covered by tests
- [x] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [x] Checked for unhandled edge cases
- [x] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [x] I'm proud of this work!

### CODE REVIEW

File: runtime/drivers/file/repo.go [hunk 1]
```diff
@@ -405,6 +414,7 @@ func (c *connection) Status(ctx context.Context, remoteBranch string) (*drivers.
 		LocalChanges:  gs.LocalChanges,
 		LocalCommits:  gs.LocalCommits,
 		RemoteCommits: gs.RemoteCommits,
+		ChangedFiles:  repoFileChanges(files),
```
Comment (AdityaHegde) [inline runtime/drivers/file/repo.go:417]: nit: the code for this vs admin/repo are slightly different. Lets make them similar?

File: web-common/src/features/project/ChangedFilesList.svelte [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      {:else}
+        <ChevronRight size="14" />
+      {/if}
+      <span>{count} {count === 1 ? "file" : "files"} changed</span>
+    </button>
+    {#if expanded}
+      <ul class="file-list">
+        {#each changedFiles as file (file.path)}
+          {@const badge =
+            badges[file.status ?? GitDiffResponse_GitFileStatus.UNSPECIFIED] ??
```
Comment (AdityaHegde) [inline web-common/src/features/project/ChangedFilesList.svelte]: This doesnt seem to be working. `file.status` is of the format `"GIT_FILE_STATUS_..."` but `GitDiffResponse_GitFileStatus` are integers, `1,2,3...`

File: web-common/src/features/project/ChangedFilesList.svelte [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    @apply flex items-center gap-x-2 text-xs;
+  }
+
+  .badge {
+    @apply flex-none text-[0.625rem] leading-none px-1 py-0.5 rounded;
+    @apply font-mono font-medium;
+  }
+
+  .badge-added {
+    @apply bg-green-100 text-green-700;
```
Comment (AdityaHegde) [inline web-common/src/features/project/ChangedFilesList.svelte]: While yellow and red have uses, green and blue do not have direct references. Maybe check Banner.svelte for colors?

The end goal is not never directly reference colors, rather aliases like `text-fg-secondary`

**Review by AdityaHegde — CHANGES_REQUESTED** [review]

File: web-common/src/features/project/ChangedFilesList.svelte (hunk 2 above)
Comment (k-anshul) [reply web-common/src/features/project/ChangedFilesList.svelte]: Thanks. I missed checking for it completely after my initial POC.

**Review by AdityaHegde — APPROVED** [review]

**Comment by nishantmonu51** [comment]
@codex: review

File: runtime/pkg/gitutil/sync.go [hunk 4]
```diff
@@ -81,6 +101,86 @@ func Status(ctx context.Context, path, subpath, remoteName, remoteBranch string)
...
+	if compareBranch == "" {
+		branch, err := Run(ctx, path, "rev-parse", "--abbrev-ref", "HEAD")
+		if err != nil {
+			return nil, err
+		}
+		compareBranch = branch
+	}
+	ref := fmt.Sprintf("%s/%s", remoteName, compareBranch)
+
+	diffArgs := []string{"diff", "--name-status", "-M", ref}
```
[the remark this answers is not in the document]
Comment (k-anshul) [reply runtime/pkg/gitutil/sync.go]: Using `merge-base`

**Review by nishantmonu51 — APPROVED** [review]
